timelock-impl/src/main/java/com/palantir/atlasdb/timelock/management/AllNodesDisabledNamespacesUpdater.java

```diff
@@ -0,0 +1,271 @@
+        return responses;
+    }
+
+    private SingleNodeUpdateResponse updateLocallyOrCheckState(
+            List<SingleNodeUpdateResponse> responses,
+            Set<Namespace> namespaces,
+            UUID lockId,
+            Supplier<SingleNodeUpdateResponse> localUpdate) {
+        if (responses.stream().allMatch(SingleNodeUpdateResponse::getWasSuccessful)
+                && responses.size() == remoteUpdaters.size()) {
+            return localUpdate.get();
+        } else {
+            Map<Namespace, UUID> incorrectlyLockedNamespaces =
+                    localUpdater.getIncorrectlyLockedNamespaces(namespaces, lockId);
+            return SingleNodeUpdateResponse.of(false, incorrectlyLockedNamespaces);
+        }
+    }
+
+    private boolean updateWasSuccessfulOnAllNodes(List<SingleNodeUpdateResponse> responses) {
+        return updateWasSuccessfulOnAllNodes(responses, clusterSize);
+    }
+
+    private static boolean updateWasSuccessfulOnAllNodes(
+            List<SingleNodeUpdateResponse> responses, int expectedResponseCount) {
+        if (responses.size() < expectedResponseCount) {
+            log.error(
+                    "Failed to reach some node(s) during update",
+                    SafeArg.of("responseCount", responses.size()),
+                    SafeArg.of("expectedResponseCount", expectedResponseCount));
+            return false;
+        }
+
+        return responses.stream().allMatch(SingleNodeUpdateResponse::getWasSuccessful);
+    }
+
+    private static Set<Namespace> getConsistentFailures(List<SingleNodeUpdateResponse> responses) {
+        return getConsistentFailures(getNamespacesByFailureCount(responses), responses.size());
+    }
+
+    private static Set<Namespace> getConsistentFailures(
+            Map<Namespace, UpdateFailureRecord> failedNamespaces, int responseCount) {
+        return KeyedStream.stream(failedNamespaces)
+                .filter(val ->
+                        val.failureCount() == responseCount && val.lockIds().size() == 1)
+                .keys()
+                .collect(Collectors.toSet());
+    }
+
+    private static Map<Namespace, UpdateFailureRecord> getNamespacesByFailureCount(
+            List<SingleNodeUpdateResponse> responses) {
+        Map<Namespace, UpdateFailureRecord> failuresByNamespace = new HashMap<>();
+        for (SingleNodeUpdateResponse response : responses) {
+            KeyedStream.stream(response.getLockedNamespaces()).forEach((namespace, lockId) -> {
+                failuresByNamespace.merge(namespace, UpdateFailureRecord.of(lockId), UpdateFailureRecord::merge);
+            });
+        }
+        return failuresByNamespace;
+    }
+
+    // Execution
+    private <T> List<T> attemptOnAllRemoteNodes(Function<DisabledNamespacesUpdaterService, T> request) {
+        Function<DisabledNamespacesUpdaterService, WrappedPaxosResponse<T>> composedFunction =
+                service -> new WrappedPaxosResponse<>(request.apply(service));
```

Comment:
It's a bit strange that here we wrap a response into an always successful paxos response, when the response itse;f can say it was unsuccessful. Maybe just use that to determine success or failure?

Reply:
I think this was a hangover from when we had separate response objects in this code path, but I wanted to extract something fancy like this.

Reply:
oh, it's because we need something that extends PaxosResponse in order to use PaxosQuorumChecker :-/

Reply:
fixed by making SingleNodeUpdateResponse extend PaxosResponse and removing the wrapper.